relay/kusama/src/impls.rs

```diff
@@ -0,0 +1,178 @@
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::xcm_config;
+use frame_support::pallet_prelude::DispatchResult;
+use frame_system::RawOrigin;
+use kusama_runtime_constants::currency::*;
+use parity_scale_codec::{Decode, Encode};
+use primitives::Balance;
+use runtime_common::identity_migrator::{OnReapIdentity, WeightInfo};
+use sp_std::{marker::PhantomData, prelude::*};
+use xcm::{latest::prelude::*, VersionedLocation, VersionedXcm};
+use xcm_executor::traits::TransactAsset;
+
+/// A type containing the encoding of the People Chain pallets in its runtime. Used to construct any
+/// remote calls. The codec index must correspond to the index of `IdentityMigrator` in the
+/// `construct_runtime` of the remote chain.
+#[derive(Encode, Decode)]
+enum PeopleRuntimePallets<AccountId: Encode> {
+	#[codec(index = 248)]
+	IdentityMigrator(IdentityMigratorCalls<AccountId>),
+}
+
+/// Call encoding for the calls needed from the Identity Migrator pallet.
+#[derive(Encode, Decode)]
+enum IdentityMigratorCalls<AccountId: Encode> {
+	#[codec(index = 1)]
+	PokeDeposit(AccountId),
+}
+
+/// Type that implements `OnReapIdentity` that will send the deposit needed to store the same
+/// information on a parachain, sends the deposit there, and then updates it.
+pub struct ToParachainIdentityReaper<Runtime, AccountId>(PhantomData<(Runtime, AccountId)>);
+impl<Runtime, AccountId> ToParachainIdentityReaper<Runtime, AccountId> {
+	/// Calculate the balance needed on the remote chain based on the `IdentityInfo` and `Subs` on
+	/// this chain. The total includes:
+	///
+	/// - Identity basic deposit
+	/// - `IdentityInfo` byte deposit
+	/// - Sub accounts deposit
+	/// - 2x existential deposit (1 for account existence, 1 such that the user can transact)
+	fn calculate_remote_deposit(bytes: u32, subs: u32) -> Balance {
+		// Remote deposit constants. Parachain uses `deposit / 100`
+		// Source:
+		// https://github.com/polkadot-fellows/runtimes/blob/55e97c5/system-parachains/constants/src/kusama.rs#L29
+		//
+		// Parachain Deposit Configuration:
+		//
+		// pub const BasicDeposit: Balance = deposit(1, 17);
+		// pub const ByteDeposit: Balance = deposit(0, 1);
+		// pub const SubAccountDeposit: Balance = deposit(1, 53);
+		// pub const EXISTENTIAL_DEPOSIT: Balance = constants::currency::EXISTENTIAL_DEPOSIT / 10;
+		let para_basic_deposit = deposit(1, 17) / 100;
```

Comment:
Because the data structure of `IdentityInfo` has changed, so a null Identity (all `Data::None`) has a different storage footprint than the data structure on the parachain.